README.md
# riemann-cookbook

TODO: Enter the cookbook description here.

## Supported Platforms

TODO: List your supported platforms.

## Attributes

<table>
  <tr>
    <th>Key</th>
    <th>Type</th>
    <th>Description</th>
    <th>Default</th>
  </tr>
  <tr>
    <td><tt>['riemann']['bacon']</tt></td>
    <td>Boolean</td>
    <td>whether to include bacon</td>
    <td><tt>true</tt></td>
  </tr>
</table>

## Usage

### riemann::default

Include `riemann` in your node's `run_list`:

```json
{
  "run_list": [
    "recipe[riemann::default]"
  ]
}
```

## License and Authors

Author:: YOUR_NAME (<YOUR_EMAIL>)


CHANGELOG.md
# 0.1.0

Initial release of riemann


metadata.rb
name             'riemann'
maintainer       'Pavel Bakhmetev'
maintainer_email 'pavelbakh@yahoo.com'
license          'All rights reserved'
description      'Installs/Configures riemann server and dashboard'
long_description 'Installs/Configures riemann server and dashboard'
version          '1.0.2'

supports "Centos", ">= 6.5"

depends "build-essential"
depends "yum-epel"
depends "iptables-ng"

recipe "riemann::server", "Riemann server"
recipe "riemann::dashboard", "Riemann dashboard"


attribute 'riemann/server/version',
  :display_name => "Server version",
  :description => "Riemann server version to install",
  :type => "string",
  :default => "0.2.9"

attribute 'riemann/server/user',
  :display_name => 'Riemann system user',
  :description => 'Name of the riemann server system user',
  :type => 'string',
  :default => 'riemann'

attribute 'riemann/server/home',
  :display_name => 'Riemann server home',
  :description => 'Home directory for the riemann server user',
  :type => 'string',
  :default => '/var/lib/riemann'

attribute 'riemann/server/config_directory',
  :display_name => 'Riemann config directory',
  :description => 'Where to put riemann.config. Should not be modified but can be useful for external cookbook to put a custom riemann.config',
  :type => 'string',
  :default => '/etc/riemann'

attribute 'riemann/server/bind_ip',
  :display_name => 'Riemann server bind address',
  :description => 'The IP address to bind riemann server on',
  :type => 'string',
  :default => '0.0.0.0'

attribute 'riemann/server/listen_port',
  :display_name => 'Riemann listen port',
  :type => 'string',
  :default => '5555'

attribute 'riemann/server/websocket_port',
  :display_name => 'Riemann websocket port',
  :type => 'string',
  :default => '5556'

attribute 'riemann/dashboard/port',
  :display_name => 'Riemann dashboard TCP port',
  :type => 'string',
  :default => '4567'

attribute 'riemann/dashboard/bind_ip',
  :display_name => 'Riemann dashboard bind address',
  :type => 'string',
  :default => '0.0.0.0'

attribute 'riemann/dashboard/user',
  :display_name => 'Riemann dashboard user',
  :description => 'Riemann dashboard system user name',
  :type => 'string',
  :default => 'riemann-dash'

attribute 'riemann/dashboard/home',
  :display_name => 'Riemann dash install dir',
  :description => 'The place where the riemann-dash bundle and configuration will be installed.',
  :type => 'string',
  :default => '/var/lib/riemann-dash'


recipes/server.rb
#
# Cookbook Name:: riemann
# Recipe:: server
#
# Copyright (C) 2015 SoftDirect Ltd
# Copyright (C) 2015 Pavel Bakhmetev
#
# All rights reserved - Do Not Redistribute
#

iptables_ng_rule 'riemann' do
  rule ["--protocol tcp --dport #{node['riemann']['server']['listen_port']} --match state --state NEW --jump ACCEPT",
        "--protocol udp --dport #{node['riemann']['server']['listen_port']} --match state --state NEW --jump ACCEPT",
        "--protocol tcp --dport #{node['riemann']['server']['websocket_port']} --match state --state NEW --jump ACCEPT",
        "--protocol udp --dport #{node['riemann']['server']['websocket_port']} --match state --state NEW --jump ACCEPT"]
end

user node['riemann']['server']['user'] do
  home node['riemann']['server']['home']
  shell "/bin/bash"
  system true
end

# Installs riemann package:
if platform?("redhat", "centos", "fedora", "amazon", "scientific")
  include_recipe "yum-epel"

  remote_file "/tmp/riemann-#{node[:riemann][:server][:version]}-1.noarch.rpm" do
    source "http://aphyr.com/riemann/riemann-#{node[:riemann][:server][:version]}-1.noarch.rpm"
    mode 0644
  end

#  gem_package "riemann-tools"
  yum_package "java-1.7.0-openjdk"
  yum_package "daemonize"

  package "riemann" do
    action :install
    source "/tmp/riemann-#{node[:riemann][:server][:version]}-1.noarch.rpm"
    provider Chef::Provider::Package::Rpm
  end

end

directory "/var/log/riemann/" do
  mode 0755
  owner node['riemann']['server']['user']
  group node['riemann']['server']['user']
  action :create
end

directory "/usr/lib/riemann" do
  mode 0775
  owner 'riemann'
  group 'riemann'
  action :create
end

service "riemann" do
  supports :restart => true
  action [:enable, :start]
end

template "/etc/riemann/riemann.config" do
  source "server/riemann.config.erb"
  owner "root"
  group "root"
  mode 0644

  notifies :restart, resources(:service => 'riemann')
  variables :bind_ip => node[:riemann][:server][:bind_ip]
end


recipes/dashboard.rb
#
# Cookbook Name:: riemann
# Recipe:: dashboard
#
# Copyright (C) 2015 SoftDirect Ltd
# Copyright (C) 2015 Pavel Bakhmetev
#
# All rights reserved - Do Not Redistribute
#

include_recipe 'build-essential::default'

iptables_ng_rule 'riemann-dash' do
  rule ["--protocol tcp --dport #{node['riemann']['dashboard']['port']} --match state --state NEW --jump ACCEPT"]
end

user node['riemann']['dashboard']['user'] do
  home node['riemann']['dashboard']['home']
  shell "/bin/bash"
  system true
end

directory node['riemann']['dashboard']['home'] do
  user node['riemann']['dashboard']['user']
end

[ 'ruby', 'ruby-devel', 'rubygems' ].each do |pkg|
  package pkg do
    action :install
  end
end

gem_package 'bundler' do
  action :install
end

cookbook_file "#{node['riemann']['dashboard']['home']}/Gemfile" do
  source "dashboard/Gemfile"
  mode "0755"
end

template "/etc/init.d/riemann-dash" do
  source "dashboard/riemann-dash.erb"
  owner "root"
  group "root"
  mode "0711"
end

execute 'bundle install --path vendor/bundle' do
  cwd node['riemann']['dashboard']['home']
  user node['riemann']['dashboard']['user']
  not_if 'bundle check'
end

template "#{node['riemann']['dashboard']['home']}/config.yml" do
  source "dashboard/config.yml.erb"
  owner node['riemann']['dashboard']['user']
  mode '0644'
end

template "#{node['riemann']['dashboard']['home']}/config.ru" do
  source "dashboard/config.ru.erb"
  owner node['riemann']['dashboard']['user']
end

service "riemann-dash" do
  supports :restart => true, :start => true, :stop => true, :reload => false
  action [:enable, :start]
end
